src/Toolkit/Toolkit/UI/Controls/PopupViewer/TextPopupElementView.cs

```diff
@@ -70,6 +74,41 @@ public override void OnApplyTemplate()
         /// </summary>
         public static readonly DependencyProperty ElementProperty =
             PropertyHelper.CreateProperty<TextPopupElement, TextPopupElementView>(nameof(Element), null, (s, oldValue, newValue) => s.OnElementPropertyChanged());
+
+        /// <summary>
+        /// Occurs when an URL is clicked.
+        /// </summary>
+        /// <remarks>
+        ///  <para>Override this to prevent the default open action.</para>
+        /// </remarks>
+        /// <param name="uri">Clicked URL</param>
+        public virtual async void OnHyperlinkClicked(Uri uri)
+        {
+            if (uri is not null)
+            {
+                var viewer = GetPopupViewerParent();
+                if (viewer is not null)
+                {
+                    var handled = viewer.OnHyperLinkClicked(uri);
+                    if (handled)
+                        return;
+                }
+
+                try
+                {
+#if WPF
+                    await Launcher.LaunchUriAsync(uri);
+#elif MAUI
+                    await Browser.OpenAsync(uri, BrowserLaunchMode.SystemPreferred);
+#endif
```

Comment:
This needs to use `Esri.ArcGISRuntime.Toolkit.Internal.Launcher.LaunchUri` which already has the cross-platform code to deal with launching. Also Line 100 doesn't work on .NET Framework.